service/history/queues/executable_factory.go

```diff
@@ -78,6 +71,8 @@ func NewExecutableFactory(
 	clusterMetadata cluster.Metadata,
 	logger log.Logger,
 	metricsHandler metrics.Handler,
+	dlqWriter *DLQWriter,
+	dlqEnabled dynamicconfig.BoolPropertyFn,
 ) *executableFactoryImpl {
```

Comment:
I don't see much difference between this func and `func (f *QueueFactoryBase) NewExecutableFactory()`

Reply:
Yeah, it's just one param so not worth it. I've inlined the function.